### PR TITLE
Fix `clojure.core` function highlighting on ClojureScript

### PR DESCRIPTION
**Pull request by vemv**
Part of https://github.com/clojure-emacs/cider/issues/3385

(Needs a trivial cider-nrepl fix too, but I'm adding this as a changelog anyway. Will be upgraded in the following few days)

### CODE REVIEW

File: cider-resolve.el [hunk 1]
```diff
@@ -107,7 +107,8 @@ This will be clojure.core or cljs.core depending on the return value of the
 function `cider-repl-type'."
   (when-let* ((repl (cider-current-repl)))
     (with-current-buffer repl
-      (cider-resolve--get-in (if (eq cider-repl-type 'cljs)
+      (cider-resolve--get-in (if (or (eq cider-repl-type 'cljs)
+                                     (derived-mode-p 'clojurescript-mode))
```
Comment (bbatsov) [inline cider-resolve.el:111]: Looking at the code this shouldn't be needed - notice the use of `with-current-buffer`, which means that will always come from the REPL and you can never really have `clojurescript-mode` there.

Comment (vemv) [reply cider-resolve.el:111]: Good eye. Then it should be a matter of simply updating cider-nrepl.

Comment (bbatsov) [reply cider-resolve.el:111]: Yep.

**Comment by vemv** [comment]
I'll just move the changelog entry to https://github.com/clojure-emacs/cider/pull/3379

**Pull request closed without merging**